cmd/api.go

```diff
@@ -286,6 +292,123 @@ var conv = &cobra.Command{
 	},
 }
 
+var limitConv = &cobra.Command{
+	Use:     "limitcvt <ECAddress> <FA-SOURCE> <SRC-ASSET> <AMOUNT> <DEST-ASSET> <LIMIT>",
+	Aliases: []string{"lcvt", "limitconversion", "limitconvert"},
+	Short:   "Builds and submits a pegnet conversion when the specified limit is reached",
+	Long: "Only converts an asset when the limit condition is met. By default this condition is " +
+		"when the source asset is below the provided limit.\nTo convert when the destination asset reaches the limit instead use the '--dest' flag.\n" +
+		"To convert when the selected asset is above the limit threshold rather than below, use the '--above' flag.\n" +
+		"to test without actually going through with the conversion use the '--dryrun' flag, it will print out when the condition is met" +
+		"",
+	Example: "pegnetd limitcvt EC3eX8VxGH64Xv3NFd9g4Y7PxSMnH3EGz5jQQrrQS8VZGnv4JY2K FA32xV6SoPBSbAZAVyuiHWwyoMYhnSyMmAHZfK29H8dx7bJXFLja" +
+		" pXBT 1 pUSD 9000.00 --above",
+	PersistentPreRun: always,
+	PreRun:           SoftReadConfig,
+	Args: CombineCobraArgs(
+		CustomArgOrderValidationBuilder(
+			true,
+			ArgValidatorECAddress,
+			ArgValidatorFCTAddress,
+			ArgValidatorAssetOrP,
+			ArgValidatorFCTAmount,
+			ArgValidatorAssetOrP,
+			ArgValidatorFCTAmount), // Limit is converted into a decimal object, using ArgValidatorFCTAmount for validation
+	),
+	Run: func(cmd *cobra.Command, args []string) {
+		var err error
+		cl := node.FactomClientFromConfig(viper.GetViper())
+		payment, source, srcAsset, amt, destAsset, limitStr := args[0], args[1], args[2], args[3], args[4], args[5]
+
+		// Let's check the pXXX -> pFCT first
+		status := getStatus()
+		if (destAsset == "pFCT" || destAsset == "FCT") && uint32(status.Current) >= node.OneWaypFCTConversions {
+			cmd.PrintErrln(fmt.Sprintf("pXXX -> pFCT conversions are not allowed since block height %d. If you need to acquire pFCT, you have to burn FCT -> pFCT", node.OneWaypFCTConversions))
+			os.Exit(1)
+		}
+
+		dest, _ := cmd.Flags().GetBool("dest")
+		above, _ := cmd.Flags().GetBool("above")
+		dryrun, _ := cmd.Flags().GetBool("dryrun")
+		waiting := true
+		limit, err := decimal.NewFromString(limitStr)
```

Comment:
Not sure why you need to import a completely new library for this. We're using `FactoidToFactoshi()` to achieve this conversion